app/Http/Controllers/Auth/LoginController.php

```diff
@@ -0,0 +1,143 @@
+        }
+        return $this->vatusaAuth($resourceOwner, $accessToken);
+    }
+
+    protected function vatusaAuth($resourceOwner, $accessToken) {
+        $client = new Client();
+        $result = $client->request('GET', 'https://api.vatusa.net/v2/user/' . $resourceOwner->data->cid . '?apikey=' . Config::get('vatusa.api_key'));
+        if ($result) {  // VATUSA API response
+            $resu = json_decode($result->getBody()->__toString(), true);
+            if (isset($resu['data'])) { // VATUSA API responded with JSON in expected format
+                $res = $resu['data'];
+                $userstatuscheck = User::find($res['cid']);
+                if ($userstatuscheck) { // User's CID found on the facility roster
+                    if ($userstatuscheck->status != 2) { // User is not marked inactive on the facility roster
+                        $userstatuscheck->fname = $res['fname'];
+                        $userstatuscheck->lname = $res['lname'];
+                        $userstatuscheck->email = $res['email'];
+                        $userstatuscheck->rating_id = $res['rating'];
+                        if ($res['flag_broadcastOptedIn'] == 1) {
+                            if ($userstatuscheck->opt != 1) {
+                                $opt = new Opt;
+                                $opt->controller_id = $res['cid'];
+                                $opt->ip_address = '0.0.0.0';
+                                $opt->means = 'VATUSA API';
+                                $opt->option = 1;
+                                $opt->save();
+                                $userstatuscheck->opt = 1;
+                            }
+                        } else {
+                            $user_opt = Opt::where('controller_id', $userstatuscheck->id)->where('means', '!=', 'VATUSA API')->where('option', 1)->first();
+                            if ($userstatuscheck->opt != 0 && !isset($user_opt)) {
+                                $opt = new Opt;
+                                $opt->controller_id = $res['cid'];
+                                $opt->ip_address = '0.0.0.0';
+                                $opt->means = 'VATUSA API';
+                                $opt->option = 0;
+                                $opt->save();
+                                $userstatuscheck->opt = 0;
+                            }
+                        }
+                        if ($userstatuscheck->visitor == '1') {
+                            if ($res['facility'] != 'ZZN') {
+                                $userstatuscheck->visitor_from = $res['facility'];
+                            }
+                        } else {
+                            $userstatuscheck->visitor_from = null;
+                        }
+                        $userstatuscheck->save();
+                        $this->completeLogin($resourceOwner);
+                        Auth::loginUsingId($res['cid'], true);
```

Comment:
This should work but these are redundant. We've already created the user (`$userstatuscheck`) and then we login with `Auth::loginUsingId($res['cid'], true);` so `$this-completeLogin($resourceOwner)` can probably just be removed. Also it's probably better if we make this change:

```diff
- Auth::loginUsingId($res['cid'], true);
+ Auth::loginUsingId($userstatuscheck->id, true);
```

Reply:
Agreed - fixed that